Makefile.arch

```diff
@@ -14,7 +14,7 @@
 
 # Name of MPI compiler and its flags:
 CMP = mpic++
-CXXFLAGS = -O3 -std=c++0x -Wall
+CXXFLAGS = -O3 -std=c++17 -Wall
```

Comment:
Do we actually use any c++17 features here? Or is this change just to bring our overall compilation flags onto a similar level?

Reply:
That was just "let's be modern" and "is this causing my issues?".